geomesa-tools/src/main/scala/org/locationtech/geomesa/tools/ingest/AutoIngestDelimited.scala

```diff
@@ -0,0 +1,171 @@
+/***********************************************************************
+* Copyright (c) 2013-2016 Commonwealth Computer Research, Inc.
+* All rights reserved. This program and the accompanying materials
+* are made available under the terms of the Apache License, Version 2.0
+* which accompanies this distribution and is available at
+* http://www.opensource.org/licenses/apache2.0.php.
+*************************************************************************/
+
+package org.locationtech.geomesa.tools.ingest
+
+import java.io.{Closeable, InputStream, InputStreamReader}
+
+import org.apache.commons.csv.{CSVFormat, CSVParser, QuoteMode}
+import org.apache.hadoop.fs.{Path, Seekable}
+import org.apache.hadoop.mapreduce.lib.input.FileInputFormat
+import org.apache.hadoop.mapreduce.{Job, TaskAttemptContext}
+import org.geotools.data.DataStore
+import org.locationtech.geomesa.features.ScalaSimpleFeature
+import org.locationtech.geomesa.jobs.mapreduce.{FileStreamInputFormat, FileStreamRecordReader}
+import org.locationtech.geomesa.tools.Utils.Formats
+import org.locationtech.geomesa.tools.Utils.Formats.Formats
+import org.locationtech.geomesa.utils.geotools.SimpleFeatureTypes
+import org.opengis.feature.simple.SimpleFeature
+
+import scala.collection.JavaConversions._
+
+/**
+ * These classes operate on files in a specific format. The format is used by the
+ * geomesa tools export, for convenience.
+ *
+ * The first line of each file is expected to be a delimited (tab or comma) list
+ * of attributes for a simple feature type, starting with the reserved word 'id'.
+ * For example:
+ *
+ * id,name:String,dtg:Date,geom:Point:srid=4326
+ *
+ * The first line of the file will be used to create a simple feature type.
+ * All subsequent lines must be type convertible to the appropriate simple feature attribute.
```

Comment:
What do you think about having the user provide the spec as before on the command line to --spec or -s and then that just lines up with the existing csv row (I guess it would be a little annoying if you need to skip header). Thought pattern being the user has a CSV file and does need to edit the CSV file (it may be in S3 or HDFS already for example) in order to AutoIngest it.